chain/net/net.go

```diff
@@ -0,0 +1,176 @@
+)
+
+// Net represents an alignment block built off of chains, it represents the UCSC net file type
+type Net struct {
+	TName       string
+	Level       int
+	Class       string
+	TStart      int
+	TSize       int
+	QName       string
+	Orientation bool
+	QStart      int
+	QSize       int
+	ExtraFields string
+	spaces      int //used for writing
+}
+
+// Read parses a net file and returns a slice of Net and a map with chromosome names and sizes in the net file
+func Read(filename string) ([]Net, map[string]chromInfo.ChromInfo) {
+	var currTName string
+	var done bool
+	var block Net
+	var currLvl [][]int = [][]int{{0, 0}, {0}} //{level, spaces}, {lvlKey}
+	mp := make(map[string]chromInfo.ChromInfo)
+	file := fileio.EasyOpen(filename)
+	var answer []Net
+	defer file.Close()
+	for block, currTName, currLvl, done = NextNet(file, currTName, currLvl, mp); !done; block, currTName, currLvl, done = NextNet(file, currTName, currLvl, mp) {
+		if block.Class != "" {
+			answer = append(answer, block)
+		}
+	}
+	return answer, mp
+}
+
+// NextNet is a helper function for the Read function. It reads lines from an EasyReader and returns a Net struct, the
+// current chromosome, and a bool which will return true once it has reached the end of the file
+func NextNet(reader *fileio.EasyReader, currTName string, currLvl [][]int, mp map[string]chromInfo.ChromInfo) (Net, string, [][]int, bool) {
+	line, done := fileio.EasyNextRealLine(reader)
+	if done {
+		return Net{}, currTName, currLvl, true
+	}
+	n, currTName, currLvl := NewNet(line, currTName, currLvl, mp)
+	return n, currTName, currLvl, false
+}
+
+// NewNet is a helper function for the Read function. It takes in a single line of the file representing a Net entry
+// and returns a filled in Net struct and the current chromosome
+func NewNet(text string, currTName string, currLvl [][]int, mp map[string]chromInfo.ChromInfo) (Net, string, [][]int) {
+	data := strings.Split(text, " ")
+	if data[0] == "net" {
+		currTName = data[1]
+		mp[data[1]] = chromInfo.ChromInfo{Name: data[1], Size: parse.StringToInt(data[2])}
+		return Net{}, currTName, [][]int{{0, 0}, {0}}
+	}
+	currLvl, newData := determineLevel(data, currLvl)
+	curr := Net{
+		TName:       currTName,
+		Level:       currLvl[0][0],
+		Class:       newData[0],
+		TStart:      parse.StringToInt(newData[1]),
+		TSize:       parse.StringToInt(newData[2]),
+		QName:       newData[3],
+		Orientation: parse.StringToStrand(newData[4]),
+		QStart:      parse.StringToInt(newData[5]),
+		QSize:       parse.StringToInt(newData[6]),
+		ExtraFields: strings.Join(newData[7:], " "),
+		spaces:      currLvl[0][1],
+	}
+
+	return curr, currTName, currLvl
+}
+
+// determineLevel is a helper struct for the NewNet function. It parses how many spaces proceed the data on the line
+// of the file to determine the Level of the net. It also returns the data without the proceeding spaces for easy parsing
+func determineLevel(data []string, currLvl [][]int) ([][]int, []string) {
```

Comment:
I would consider creating a new struct to track the levels:
- but that may require a lot of refactoring so I'll leave that decision to you. 

```golang
type LevelInfo struct {
    Level  int
    Spaces int
    LvlKey []int
}
```

Reply:
yes i like this a lot. It was one of those things where i thought it was going to be simple, but i kept having to add complexity. Best to make a struct for readability